src/core/types/menu.ts

```diff
@@ -3,6 +3,7 @@ export type MenuItem = MenuItemWithLink | MenuItemWithChildren
 export interface MenuItemWithLink {
   text: string
   link: string
+  badge?: string | MenuBadgeItem
```

Comment:
string is as required type when creating MenuBadgeItem

why not: 
badge?: MenuBadgeItem

Reply:
this is string OR MenuBadgeItem
I made it that way to let use both string(in case tyep is default) and object in configs
so you can add
```
 {
    text: 'Experts',
    badge: 'NEW',
    ......
```
OR
```
 {
    text: 'Experts',
    badge: {
       text: 'NEW',
       type: 'warning'
    },
    ......
```

Reply:
I do get OR, but I don't see a value. If you are specifying a badge, you need to always define a string or text, and the only optional thing is type.

I am seeing more code that is not clean having this OR statement.

if you want to use badge you need to minimum {text: "badge name"}